### ISSUE TITLE
Elasticsearch container failing SSL on Apple Silicon

### ISSUE THREAD

**Issue by shiddy** [sherifabdlnaby/elastdocker#54]
**Describe the bug**
I've attempted pulling and running this with Docker Desktop on an x86 and a M1 mac machines from master and a clean docker cache (`docker system prune -a`).

On the M1 machine, The 3 containers spun up with `make setup && make elk` will be created without issue. However the elasticsearch container will always be in a (starting) state.

![image](https://user-images.githubusercontent.com/3847723/149168412-b2c481b8-04f7-4207-be39-0ee3627211cc.png)

I am able to pull logs from the docker-compose command for elasticsearch, and have pasted them here:
https://pastebin.com/eFB33LEQ

When I attempt to access the elasticsearch node via the port 9200 I recieve the following:

`curl -k https://127.0.0.1:9200`
`curl: (7) Failed to connect to 127.0.0.1 port 9200: Connection refused` OR `curl: (35) LibreSSL SSL_connect: SSL_ERROR_SYSCALL in connection to 127.0.0.1:9200`

**To Reproduce**
1) pull elastdocker on M1 machine
2) run `make setup` and wait for completion
3) run `make elk`
4) run `curl -k https://127.0.0.1:9200` and `docker ps`

**Expected behavior**
I would expect that elasticsearch is able to begin it's service with the self-signed cert from make setup

**Screenshots**
![image](https://user-images.githubusercontent.com/3847723/149172010-a3e1ed87-73c8-497a-8607-fb4647815236.png)

**Desktop (please complete the following information):**
 - Device: Mac Mini m1 chip & macbook pro m1
 - OS: macOS big sur 11.5.2
 - Version Docker version 20.10.11, build dea9396 & docker-compose version 1.29.2, build 5becea4c

**Additional context**

Full log dump from elasticsearch: https://pastebin.com/eFB33LEQ

**Comment by shiddy** [comment]
I also should point out that it remains like this indefinitely on the M1, whereas this SSL response can occasionally show up on x86 but tends to resolve itself ~2 mins after starting the containers.

**Comment by sherifabdlnaby** [comment]
@shiddy Can't test Elastdocker setup with M1 as I don't have M1 Macs at my disposal. However, the error is coming from curl, could it be a cURL's problem ? Have you tried going to https://localhost:9200 via Browser?

**Comment by shiddy** [comment]
For sure, I have attempted accessing it from a browser and Firefox, Chrome, and Safari all return ssl issues or no data received.

I've attempted curl with the '-k' argument to ignore SSL. Which when using an x86 does return json data from elasticsearch.

Happy to hop on any call to help debug. I recognize it's hard to get context without more hands on keyboard access.

**Comment by sherifabdlnaby** [comment]
@shiddy Yea, it will be hard to debug without M1 Access, have you tried asking docker not to use ARM Images and use x86+Rosetta instead? I believe it tries to get ARM Images by default, am I right?

**Comment by shiddy** [comment]
Just reaching out to follow up,  I am happy to dig in more with my machine though I am a little lost as to what to test next

**Comment by sherifabdlnaby** [comment]
@shiddy Hey, sorry, haven't been able to find time to look into this. I don't even have a theory, to be honest, what happens if you try curl **from inside the container ?**, can you try exec-ing inside and see if HTTPS is responding?

**Comment by shiddy** [comment]
I think that I finally figured it out. Remarkably silly conclusion.

One of the DIMMS on my M1 was failing memtest. During my regular usage of my mac, I didn't use enough resources to use 4GB of memory at a time. After finding that I had this memory issue and getting my M1 fixed and passing memtest and having my docker desktop VM with 4GB of memory I was able to run `make setup` and `make elk` without issues, even without changing the platform argument for the image definitions.

TL;DR check the ram on your machine when you get new hardware.

**Comment by sherifabdlnaby** [comment]
@shiddy Wow, that would have been too hard to debug 😅 Thank you for coming back and posting the conclusion.
